File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.java
class Solution {
    public String shortestBeautifulSubstring(String s, int k) 
    {
        int p = 0;
        int c = 0;
        int start = 0;
        String A[] = new String [s.length()];
        for (int i = 0; i < s.length(); i++)
        {
            if (s.charAt(i)=='1')
            {
                c++;
            }
            if (c == k)
            {
                while (s.charAt(start) == '0') 
                {
                    start++;
                }
                A[p] = s.substring(start, i+1);
                p++;
                c--;
                start++;
            }
        }
        for (int i = 0; i < p; i++)
        {
            for (int j = 0; j < p - i - 1; j++)
            {
                if (A[j].length() > A[j + 1].length() ||
                    (A[j].length() == A[j + 1].length() &&
                     A[j].compareTo(A[j + 1]) > 0)) 
                {
                    String temp = A[j];
                    A[j] = A[j + 1];
                    A[j + 1] = temp;
                }
            }
        }
        if (p == 0)
        {
            return "";
        }
        return (A[0]);
    }
}

File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s, k):
        p = 0
        c = 0
        start = 0
        A = [None] * len(s)
        for i in range(len(s)):
            if s[i] == '1':
                c += 1
            if c == k:
                while s[start] == '0':
                    start += 1
                A[p] = s[start:i + 1]
                p += 1
                c -= 1
                start += 1
        for i in range(p):
            for j in range(p - i - 1):
                if (len(A[j]) > len(A[j + 1]) or
                    (len(A[j]) == len(A[j + 1]) and
                     A[j] > A[j + 1])):
                    temp = A[j]
                    A[j] = A[j + 1]
                    A[j + 1] = temp
        if p == 0:
            return ""
        return A[0]